bootstrap-joining-demo/aws-api-ecs/src/main/scala/akka/cluster/bootstrap/DemoApp.scala

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (C) 2017 Lightbend Inc. <http://www.lightbend.com>
+ */
+
+package akka.cluster.bootstrap
+
+import java.net.InetAddress
+
+import akka.actor.ActorSystem
+import akka.discovery.awsapi.ecs.AsyncEcsSimpleServiceDiscovery
+import akka.management.AkkaManagement
+import akka.management.cluster.bootstrap.ClusterBootstrap
+import com.typesafe.config.ConfigFactory
+
+object DemoApp {
+
+  def main(args: Array[String]): Unit = {
+    val privateAddress = getPrivateAddressOrExit
+    val config = ConfigFactory
+      .systemProperties()
```

Comment:
`load().withFallback` doesn't work because when the inline config defined here is the fallback, `akka.remote.netty.tcp.hostname` ends up being "" (from the akka-remote `reference.conf`), and `akka.management.http.hostname` ends up being "\<hostname\>" (from the akka-management `reference.conf`).

That results in the respective modules then attempting to call `InetAddress.getLocalHost.getHostAddress` (and failing as described in discovery.md).

So this inline config does need to have precedence over any `ConfigFactory` method which returns something including the reference configs.

 I was going for minimalism* with `ConfigFactory.systemProperties()`, but one working alternative would be `ConfigFactory.parseString("...").withFallback(ConfigFactory.load())`, if that's preferable?

*Which works because I see that `ActorSystem.scala` does this:
```
    final val config: Config = {
      val config = cfg.withFallback(ConfigFactory.defaultReference(classLoader))
      config.checkValid(ConfigFactory.defaultReference(classLoader), "akka")
      config
    }
```